.github/workflows/devshell.yml

```diff
@@ -80,7 +80,7 @@ jobs:
       - name: make check
         id: make_check
         working-directory: ./build
-        run: make V=1 check || (cat test-suite.log && false)
+        run: make V=1 check || (mkdir -p ${COREFILES_DIR} && find . -executable -a -type f | tar -cf ${COREFILES_DIR}/test-binaries.tar  --files-from=- && cat test-suite.log && false)
```

Comment:
[optional]

I would rather see a separate artifact gathering and uploading step, which runs on any kind of failure, not just on the `make check` step, but it is up to you.

Reply:
That would be a useful improvement, I agree. But I'd merge this as is at this point.